packages/@react-spectrum/dialog/src/AlertDialog.tsx

```diff
@@ -64,49 +65,61 @@ function AlertDialog(props: SpectrumAlertDialogProps, ref: DOMRef) {
     }
   }
 
+  let onKeyDown = (e) => {
+    let dialogNode: HTMLElement = dialogRef.current.UNSAFE_getDOMNode();
```

Comment:
close, but if someone passes in a ref, then dialogRef current will be undefined

instead, I think you can follow what I wrote in this comment https://github.com/adobe/react-spectrum/pull/3325/files#r939457766
let me know if it isn't working for some reason